packages/types/src/clerk.ts

```diff
@@ -472,6 +472,26 @@ export interface Resources {
 
 export type RoutingStrategy = 'path' | 'hash' | 'virtual';
 
+export type RoutingOptions =
+  | {
+      /**
+       * Page routing strategy
+       */
+      routing?: Exclude<RoutingStrategy, 'path'>;
+      /**
+       * Root URL where the component is mounted on, eg: '/sign-in'
+       */
+      path?: never;
+    }
+  | {
+      routing: 'path';
+      path: string;
+    }
+  | {
+      routing?: never;
+      path: string;
+    };
```

Comment:
Do you think this would be more readable? 

```
export type RoutingOptions =
  | { path: string }
  | { path: string; routing: Extract<RoutingStrategy, 'path'> }
  | { path?: never; routing: Extract<RoutingStrategy, 'hash' | 'virtual'> };
```

Reply:
Wouldn't this work as well?
```
export type RoutingOptions =
  | { path: string; routing?: Extract<RoutingStrategy, 'path'> }
  | { path?: never; routing?: Extract<RoutingStrategy, 'hash' | 'virtual'> };
  ```